### PR TITLE
Remove id from TransportEndpoint and rename it to Address

### PR DESCRIPTION
**Pull request by antonkharenko**

### CODE REVIEW

File: cluster/src/main/java/io/scalecube/cluster/fdetector/FailureDetectorSettings.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
+package io.scalecube.cluster.fdetector;
+
+public final class FailureDetectorSettings {
```
Comment (artem-v) [inline cluster/src/main/java/io/scalecube/cluster/fdetector/FailureDetectorSettings.java]: Better name ``FailureDetectorConfig``

File: cluster/src/main/java/io/scalecube/cluster/gossip/GossipProtocolSettings.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
+package io.scalecube.cluster.gossip;
+
+public final class GossipProtocolSettings {
```
Comment (artem-v) [inline cluster/src/main/java/io/scalecube/cluster/gossip/GossipProtocolSettings.java]: Better name ``GossipProtocolSettings``

File: transport/src/main/java/io/scalecube/transport/Address.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import javax.annotation.concurrent.Immutable;
+
+@Immutable
+public final class Address {
```
Comment (artem-v) [inline transport/src/main/java/io/scalecube/transport/Address.java]: Better name ``TransportAddress`` .  We already have ``ClusterMember`` , i.e. it wasn't not named ``Member``; same is here.

Comment (antonkharenko) [inline transport/src/main/java/io/scalecube/transport/Address.java]: Address is clear and short name; ClusterMember is a mistake - you are right - I will rename it to just Member.

File: cluster/src/main/java/io/scalecube/cluster/gossip/GossipProtocol.java [hunk 4]
```diff
@@ -48,35 +48,45 @@
 public final class GossipProtocol implements IGossipProtocol, IManagedGossipProtocol {
   private static final Logger LOGGER = LoggerFactory.getLogger(GossipProtocol.class);
 
-  private ITransport transport;
+  private final String memberId;
+  private final ITransport transport;
+  private final ScheduledExecutorService executor;
```
Comment (artem-v) [inline cluster/src/main/java/io/scalecube/cluster/gossip/GossipProtocol.java]: Always wondered why ``gossip`` does have its own executor, while ``clusterMembership`` and ``failureDetector`` don't. Can't we in this PR create dedicated executors for ``clusterMembership`` and ``failureDetector`` -s ?

File: cluster/src/main/java/io/scalecube/cluster/gossip/GossipProtocol.java [hunk 5]
```diff
@@ -48,35 +48,45 @@
...
+   *
+   * @param memberId id of current member
+   * @param transport transport
+   * @param settings gossip protocol settings
+   */
+  public GossipProtocol(String memberId, ITransport transport, GossipProtocolSettings settings) {
+    this.memberId = memberId;
+    this.transport = transport;
+    this.settings = settings;
+    this.executor = Executors.newSingleThreadScheduledExecutor(createThreadFactory("scalecube-gossip-scheduled-%s"));
```
Comment (artem-v) [inline cluster/src/main/java/io/scalecube/cluster/gossip/GossipProtocol.java]: In analogy with ``BootstrapFactory`` this ``"scalecube-gossip-scheduled-%s"`` name must be ``"sc-gossip-%s"``

File: cluster/src/main/java/io/scalecube/cluster/gossip/GossipProtocol.java (hunk 4 above)
Comment (antonkharenko) [inline cluster/src/main/java/io/scalecube/cluster/gossip/GossipProtocol.java]: It make sense, but there are more issues with executor. I think it deserves separate PR.
